x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/create_security_rule_type_wrapper.ts

```diff
@@ -200,7 +208,10 @@ export const createSecurityRuleTypeWrapper: CreateSecurityRuleTypeWrapper =
                     {
                       index: inputIndex,
                       fields: hasTimestampOverride
-                        ? ['@timestamp', timestampOverride]
+                        ? [
+                            timestampOverride,
+                            ...(shouldDisableTimestampFallback ? [] : ['@timestamp']),
```

Comment:
We have similar conditional logic based on both `timestampOverride` and `shouldDisableTimestampFallback` both here and deeper in the rule executors where we build the rule queries. I think it will simplify the query building logic to handle `shouldDisableTimestampFallback` only at this outer layer in `createSecurityRuleTypeWrapper` and create 2 constants `const primaryTimestamp: string = timestampOverride ?? '@timestamp'` and `const secondaryTimestamp: string | undefined = (timestampOverride != null && !shouldDisableTimestampFallback) ? '@timestamp' : undefined` that we pass in to the individual rule type executors.

The individual rule executors then would not be responsible for determining how to handle timestamp overrides and fallbacks, they just used the primary and secondary timestamp fields. The logic here would then become 
```
fields: secondaryTimestamp ? [primaryTimestamp, secondaryTimestamp] : [primaryTimestamp]
```

Reply:
Yeah, sounds good to me!